imapserver/imapmemserver/mailbox.go

```diff
@@ -418,13 +431,40 @@ func (mbox *MailboxView) staticSearchCriteria(criteria *imap.SearchCriteria) {
 }
 
 func (mbox *MailboxView) Store(w *imapserver.FetchWriter, numSet imap.NumSet, flags *imap.StoreFlags, options *imap.StoreOptions) error {
-	mbox.forEach(numSet, func(seqNum uint32, msg *message) {
-		msg.store(flags)
-		mbox.Mailbox.tracker.QueueMessageFlags(seqNum, msg.uid, msg.flagList(), mbox.tracker)
+	var modifiedSeqNums []uint32
+	var modifiedMsgs []*message
+
+	mbox.mutex.Lock()
+	mbox.forEachLocked(numSet, func(seqNum uint32, msg *message) {
+		if options != nil && options.UnchangedSince > 0 && msg.modSeq > options.UnchangedSince {
+			return
+		}
+
+		if changed := msg.store(flags); changed {
+			mbox.highestModSeq++
+			msg.modSeq = mbox.highestModSeq
```

Comment:
Could we handle the mod seq bump inside `msg.store`? This would make it so we don't forget bumping it if we call this function from elsewhere.

Reply:
agreed, updated